lib/cisco_node_utils/bgp.rb

```diff
@@ -486,21 +519,41 @@ def event_history_periodic
       match = config_get('bgp', 'event_history_periodic', @get_args)
       if match.is_a?(Array)
         return 'false' if match[0] == 'no '
-        return 'size_' + match[1] if match[1]
+        if Utils.nexus_evergreen?
+          return 'true' unless match[1]
+          # check if the val is size in bytes
+          return match[1] if match[1] =~ /\A\d+\z/
+          return 'size_' + match[1]
+        else
+          return 'size_' + match[1] if match[1]
+        end
       end
       default_event_history_periodic
     end
 
     def event_history_periodic=(val)
-      size = val[/small|medium|large|disable/]
+      if Utils.nexus_evergreen?
```

Comment:
Done here too. I got rid of Utils.nexus_evergreen? everywhere except the default for periodic. The get method for event_history_periodic is a bit of challenge but I got it though I am not sure if this is optimized enough. Let me know after you take another look